.travis-opam-coverage.sh

```diff
@@ -25,13 +25,37 @@ opam repository remove default
 opam repository add xs-opam https://github.com/xapi-project/xs-opam.git
 
 # install the dependencies of xapi
+sudo apt install -y libpci-dev
 opam pin add --no-action xapi .
 opam depext --yes xapi
 opam install --deps-only xapi
 
 # build and test xapi with coverage, then submit the coverage information to coveralls
-sudo apt-get install --yes wget
-wget https://raw.githubusercontent.com/simonjbeaumont/ocaml-travis-coveralls/master/travis-coveralls.sh
-COV_CONF="./configure" bash -ex travis-coveralls.sh
+
+opam install -y bisect_ppx ocveralls
+
+export BISECT_ENABLE=YES
+
+# the test fails when run on debian-stable (it passed on Ubuntu though)
+# disable it when running coverage
+# -p doesn't work to filter out subdirectories though
+# because the test in pci doesn't say which package it is part of
+rm -rf ocaml/pci/lib_test
+jbuilder runtest
+
+
+outs=$(find . | grep bisect.*.out)
```

Comment:
Is it worth to update this code to prevent issues as done in:

https://github.com/xapi-project/rrd-transport/commit/93d93c147d138c1edd9c1d367ae440e9db334523

?

Reply:
Had to use slightly different syntax to get all the values from the array, `$outs` only printed one:
```
+ outs=($(find . | grep bisect.*.out))
++ find .
++ grep 'bisect.*.out'
++ dirname ./_build/default/ocaml/xapi-types/bisect0001.out
+ bisect-ppx-report -I ./_build/default/ocaml/xapi-types -text report ./_build/default/ocaml/xapi-client/bisect0001.out
++ dirname ./_build/default/ocaml/xapi-types/bisect0001.out
+ bisect-ppx-report -I ./_build/default/ocaml/xapi-types -summary-only -text summary ./_build/default/ocaml/xapi-client/bisect0001.out
+ '[' -n '' ']'
```

Using the commit I pushed:
```
+ declare -a outs
+ outs=($(find . | grep bisect.*.out))
++ find .
++ grep 'bisect.*.out'
++ dirname ./_build/default/ocaml/xapi-types/bisect0001.out
+ bisect-ppx-report -I ./_build/default/ocaml/xapi-types -text report ./_build/default/ocaml/xapi-client/bisect0001.out ./_build/default/ocaml/xapi-types/bisect0001.out ./_build/default/ocaml/xapi/bisect0004.out ./_build/default/ocaml/xapi/bisect0001.out ./_build/default/ocaml/xapi/bisect0003.out ./_build/default/ocaml/xapi/bisect0002.out ./_build/default/ocaml/xapi/bisect0005.out
++ dirname ./_build/default/ocaml/xapi-types/bisect0001.out
+ bisect-ppx-report -I ./_build/default/ocaml/xapi-types -summary-only -text summary ./_build/default/ocaml/xapi-client/bisect0001.out ./_build/default/ocaml/xapi-types/bisect0001.out ./_build/default/ocaml/xapi/bisect0004.out ./_build/default/ocaml/xapi/bisect0001.out ./_build/default/ocaml/xapi/bisect0003.out ./_build/default/ocaml/xapi/bisect0002.out ./_build/default/ocaml/xapi/bisect0005.out
+ '[' -n '' ']'
```

Reply:
You are right, the script was broken. I've sent a PR to rrd-transport to fix it. We should consider rewriting it in a more robust way, and (as I think you suggested) store in a repo from where all the packages that need it can directly use it instead of replicating the code